lib/AstOfLlbc.ml

```diff
@@ -1604,6 +1604,17 @@ let is_str env var_id =
 
 let is_dst_var env var_id = Option.is_some (is_dst env (snd (lookup env var_id)))
 
+let rec differs_in_lifetime_only (f : C.ty) (t : C.ty) =
+  match f, t with
+  | C.TRawPtr (f, fk), C.TRawPtr (t, tk)
+  | C.TRef (_, f, fk), C.TRef (_, t, tk) -> fk = tk && differs_in_lifetime_only f t
+  | C.TArrow f, C.TArrow t ->
+    let f_args, f_ret = f.binder_value in
+    let t_args, t_ret = t.binder_value in
+    List.length f_args = List.length t_args &&
```

Comment:
That check is not needed, `for<'a> fn(&'a u32, &'a u32)` and `for<'a, 'b> fn(&'a u32, &'b u32)` are inter-convertible.